### DISCUSSION TITLE
PHP 7.4 oder PHP 8 - Ansätze um Kompatibilität zu steigern

### DISCUSSION THREAD

**Discussion by niclasd** [scheb/open-apexx#61] [General]
Hallo zusammen,

der Hoster macht wieder "Druck" und will alte PHP-Versionen abschalten.
Ich benötige in wenig Guidance, damit ich weiß in welche Richtung ich gehen muss:

- Gibt es Tools, um nicht mehr unterstützte PHP-Befehle zu erkennen, die ich anschließend durch neue Varianten ersetzen kann? (bin Coding-Laie, aber würde es mal probieren ;) )
- Gibt es eine Art "Sandbox", mit der Apexx auch in einer PHP 8 Umgebung "out of the box" läuft?

Ansonsten ist die PHP 7.4 Unterstützung möglicherweise gar nicht so kompliziert: Es funktioniert soweit alles gut (auch die Bildergalerie), lediglich innerhalb von Artikel/Newsbeiträgen wird statt dem tatsächlichen Bild der Platzhalter angezeigt (z.B. **{IMAGE(4473)}** )

In PHP 8 wird es vermutlich sportlicher. Dort erhalte ich einen 500er Server Error. Ich gehe also davon aus, dass ich in spätestens 2 Jahren irgendwie auf ein neues System migrieren sollte :-(

Vielen Dank für jede Rückmeldung!!!

**Comment by bigfraggle** [comment]
Hab letztens auch auf die 7.4 aktualisiert und eigentlich keine Fehler gefunden... Aber Tatsache,  Inline-Bilder gehen wirklich nicht mehr.

**Comment by Tropby** [comment]
Der Fehler lässt sich leicht beheben. 

In der Datei "**modules/mediamanager/system.php**" in Zeile **28** muss das letzte "**e**" im Pattern entfernt werden.

`preg_match_all('#{IMAGE\(( *[0-9]+ *)\)}#se',$text,$match);`

ändern zu 

`preg_match_all('#{IMAGE\(( *[0-9]+ *)\)}#s',$text,$match);`

**Comment by niclasd** [comment]
> Der Fehler lässt sich leicht beheben.
> 
> In der Datei "**modules/mediamanager/system.php**" in Zeile **28** muss das letzte "**e**" im Pattern entfernt werden.
> 
> `preg_match_all('#{IMAGE\(( *[0-9]+ *)\)}#se',$text,$match);`
> 
> ändern zu
> 
> `preg_match_all('#{IMAGE\(( *[0-9]+ *)\)}#s',$text,$match);`

Ach super, vielen Dank für die schnelle Hilfe :-)!
Ich glaube da wäre ich nie drauf gekommen und kann es auch nicht so ganz nachvollziehen - war das ein "Vertipper"?
